socorro/external/postgresql/connection_context.py

```diff
@@ -2,15 +2,32 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import os
 import socket
 import contextlib
 import psycopg2
 import psycopg2.extensions
+from urlparse import urlparse
 
-from configman.config_manager import RequiredConfig
-from configman import Namespace
+from configman import RequiredConfig, Namespace
+from configman.memoize import memoize
 
+#------------------------------------------------------------------------------
+# we're going to end up parsing the same url several times, there's no need to
+# actually do it more than once, so we wrap the method with a memoizer to speed
+# things along.
+@memoize()
```

Comment:
I read your comment on Solution 2. Twice. I still don't understand it but I understand that you know this domain much better than me. Legacy is definitely at play. And transition from legacy to modern is the crux. 

My dream is to have it like you do in sugardough (the mozilla preferred way of doing modern django set ups) in that the one and only way to set credentials is with a URL. 

What would it take to get rid of ALL "database_something" (where "something" is **not** "url")?